smartgrid-prediction/README.asciidoc

```diff
@@ -0,0 +1,130 @@
+# Introduction
+
+This sample showcases a fictional use of Spring XD in the context of smart electricity metering and production. It was inspired by the http://www.cse.iitb.ac.in/debs2014/?page_id=42[ACM Distributed Event Based Systems Grand Challenge 2014] and re-uses some of its sample input data.
+
+Smart plugs are used to measure power consumption and report metrics to the system. In this sample, smartplugs are simulated by data ingestion using JMeter. Also, to have the demo display some lively data, everything has been re-sampled to be played at 60x the original speed (as such, the 16hrs worth of data available are ingested in ~16mins).
+
+The ingestion, aggregation and machine learning features of Spring XD are used to display and predict the energy consumption of the grid, in a scalable way.
+
+The overall logical architecture of this sample is depicted below:
+
+image::arch.png[]
+
+# Understanding the sample
+
+You'll find several directories in this sample. Not all are needed to actually run the solution, but are included for complete reference. In addition, the main piece of the puzzle is the list of stream definitions that make up the XD solution. You'll read about them in the xref:running[] section.
+
+* link:jmeter[] contains the JMeter plan and supporting scripts that can be used to ingest data in Spring XD.
+* link:frontend[] contains a small serverside app (and accompanying frontend) that reads aggregated counters and displays them
+* link:prediction[] is the home of the link:prediction/metaModel.pmml[Neural Network models] that can predict consumption, as well as explanations about how to re-create that file from scratch.
+
+[[running]]
+# Running the demo
+## Install the analytic-pmml module
+
+This sample makes use of the [analytic-pmml] module which is not part of the main Spring XD distribution and has to be built from source. You can read the link:https://github.com/spring-projects/spring-xd-modules/tree/master/analytics-ml-pmml[README] in its repository to build and install it, or you can use the provided link:setup-pmml.sh[] script that automates that process.
+
+*In all cases*, you'll need to delete the `analytic-pmml` placeholder directory that can be found in the standard XD distribution before installing the hand-built module.
+
+[NOTE]
+.PMML Model also included in the module itself
+====
+The script referenced above also embeds the link:prediction/metaModel.pmml[Neural Network models] file directly inside the built `analytic-pmml` module. This is so that we can easily reference it later in the stream definitions as `classpath:metaModel.pmml`. *This is not something* you would typically do, but makes writing this README easier. In a real world scenario, you would reference the model file by its path on the container filesystem (something we obviously can't predict at the time of writing).
```

Comment:
I'd go for prediction.pmml - no need to have "model" in the filename at all IMO.